source/event_stream.c

```diff
@@ -1449,6 +1452,29 @@ static void s_reset_state(struct aws_event_stream_streaming_decoder *decoder) {
     decoder->state = s_start_state;
 }
 
+void aws_event_stream_streaming_decoder_init_from_options(
+    struct aws_event_stream_streaming_decoder *decoder,
+    struct aws_allocator *allocator,
+    const struct aws_event_stream_streaming_decoder_options *options) {
+    AWS_ASSERT(decoder);
+    AWS_ASSERT(allocator);
+    AWS_ASSERT(options);
+    AWS_ASSERT(options->on_error);
+    AWS_ASSERT(options->on_header);
+    AWS_ASSERT(options->on_payload_segment);
+    AWS_ASSERT(options->on_prelude);
+    AWS_ASSERT(options->on_prelude);
+
+    s_reset_state(decoder);
+    decoder->alloc = allocator;
+    decoder->on_error = options->on_error;
+    decoder->on_header = options->on_header;
+    decoder->on_payload = options->on_payload_segment;
+    decoder->on_prelude = options->on_prelude;
+    decoder->on_complete = options->on_complete;
+    decoder->user_context = options->user_data;
+}
+
 void aws_event_stream_streaming_decoder_init(
```

Comment:
trivial: modify the original function so it calls the new function, instead of having 2 copy/pasted versions of the same thing.

makes it less likely that someone in the future modifies 1 of these functions, but forgets to modify the other

Reply:
Thanks, fixed.